source/ai-server-logger.js

```diff
@@ -0,0 +1,122 @@
+(function() {
+    'use strict';
+
+    angular.module('ai.public.serverLogger', ['ng'])
+        .config(function($provide) {
+            $provide.decorator('$log', function($delegate, $injector) {
+                var logger = $injector.get('AiServerLogger');
+                var delegateError = $delegate.error;
+                var delegateWarn = $delegate.warn;
+                var delegateInfo = $delegate.info;
+                var delegateDebug = $delegate.debug;
+
+                var enhanceLoggingFunction = function(loggingFunction, severity) {
+                    return function() {
+                        var args = [].slice.call(arguments);
```

Comment:
I still like _.toArray better.  Regarding conversion to array, actually it was because of JSON.stringify.  I thing the following serialization is better for further handling on the server:
``` "Message": "[\"foobar\",{\"foo\":\"bar\"}]", ``` if I pass in arguments directly I get this serialization: ``` "Message": "{\"0\":\"foobar\",\"1\":{\"foo\":\"bar\"}}", ```